adding Pomerium as zero-trust proxy service (#41)
sighupio/fury-kubernetes-ingress: katalog/pomerium/README.md

# Pomerium Setup

> This document is intended to give a brief overview on how Pomerium can be implemented, for further details, please look at the official doc : <https://www.pomerium.io/docs/>

Reviewer: Can we please add a minimal description of what Promerium is? a one-liner should be enough, like we do for the other packages.

Please add it also to the repo's root README packages list :)

# Pomerium

Pomerium is an identity-aware proxy that enables secure access to internal applications. Pomerium provides a standardized interface to add access control to applications regardless of whether the application itself has authorization or authentication baked-in

## Pomerium Setup

This document is intended to give a brief overview on how Pomerium can be implemented, for further details, please look at the official doc : <https://www.pomerium.io/docs/>

## Setup

> the base kustomization component present [here](./kustomization.yaml) allows to quickly integrate this service with an existing Dex service that very likely is connected to an LDAP.

Reviewer 2: ```suggestion
The base kustomization file present [here](./kustomization.yaml) allows to quickly integrate this service with an existing Dex service, that could for example be connected to LDAP.
```

> For this reason you will need to edit your Dex configuration in order to add a static client for Pomerium service like in the example above:

Reviewer 2: ```suggestion
In order to do so, you will need to edit your Dex configuration, adding a static client to be used by Pomerium, like in the example below:
```

```yaml
>>staticClients:
    - id: "pomerium-auth-client"
      secret: "your-super-secret"
      name: "Pomerium"
      redirectURIs:
       - "https://pomerium.example.com/oauth2/callback"
```

> setting the redirectURIs, accordingly with the host you used for the pomerium ingress.

Reviewer 2: ```suggestion
Configure the `redirectURIs` section accordingly to the hosts used for the pomerium ingress.
```

> Once do that, you will need to use ovverride the configuration example ([policy](./config/policy.example.yaml) and environment variables as [configmap](./config/config.example.env) and [secret](secrets/pomerium.example.env)) like in the example above:

Reviewer 2: ```suggestion
Once dex is configured correctly, you will need to ovverride the configuration example ([policy](./config/policy.example.yaml) and environment variables via a [configmap](./config/config.example.env) and [secret](secrets/pomerium.example.env)) like in the example below:
```

```yaml
configMapGenerator:
  - name: pomerium-policy
    behavior: replace
    files:
      - policy.yml=config/pomerium-policy.yml
  - name: pomerium
    behavior: replace
    envs:
      - config/pomerium-config.env

secretGenerator:
  - name: pomerium-env
    behavior: replace
    envs:
      - secrets/pomerium.env
```

> just copy the examples in the module and override them accordingly with your settings.

Reviewer 2: ```suggestion
Just copy the examples in the module and override them according to your settings.
```

> **⚠ WARNING: in the policy file, you'll need to setup many policy as many ingress you want to protect under the pomerium authorization service.**

Reviewer 2: ```suggestion
**⚠ WARNING: in the policy file, you'll need to set up a policy for each ingress you want to protect with Pomerium authorization service.**
```

## Ingresses

> Once do that, Pomerium and Dex and correctly configured.
> The only missing step is to add annotations to the ingresses you've added previously in the policy yaml file:

Reviewer 2: ```suggestion
Once Pomerium and Dex are correctly configured, the last step is to add annotations to the ingresses you've added previously in the policy yaml file:
```
